[flljy940] iP

flljy940 opened the pull request:
# Rocky
> "Your mind is for having ideas, not holding them." - David Allen ([source](http://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

All you need to do is,

1. download it from [here](https://blackscreen.app/).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :grinning:

And it is **FREE**!

Feature:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

#

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Rename, Greet, Exit

Commit:
Updated Rocky.java with new content

Commit:
Echo

Commit:
Add, List

Commit:
List down tasks

Commit:
Mark as Done

Commit:
Implement classes for Todos, Events, Deadlines

Commit:
ToDos, Events, Deadlines

Commit:
Automated Text UI Testing

Commit:
Handle Errors

Commit:
Delete function

Commit:
Add documentation

Commit:
Refactor code

Commit:
Use Enums to identify command

Commit:
Fix DELETE bugs

Commit:
Add FileManager for file saving

Commit:
Fix tag field issues

Commit:
Add file support

Commit:
Fix parseFileSaveFormat index error

Commit:
Fix minor issue

Commit:
Update Deadline and Event date save format

Commit:
Rename variables

Commit:
Use more OOP

Commit:
Organize into Packages

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Add tests for TaskList

Commit:
Add ParserTest

Commit:
Fix minor issues

Commit:
Update .gitignore t exclude jar related directory

Commit:
Add javadoc in all classes

Commit:
Change method name to better describe behavior

Commit:
Add command to find Task with keyword

Commit:
Merge conflicts from branch-Level-9 to master

shuang26 commented on src/main/java/rocky/ui/Ui.java:
```diff
@@ -0,0 +1,90 @@
+    /**
+     * Logo string
+     */
+    private static final String logo =
```
Very cool logo 🔥

shuang26 commented on src/main/java/rocky/Rocky.java:
```diff
@@ -0,0 +1,180 @@
+     * @param action Command from user
+     * @throws RockyException General exception for invalid user command: invalid command, invalid arguments, etc.
+     */
+    private static void handleAction(Command action) throws RockyException {
```
The logic in handling the user input is clear👍 
I guess the only improvement is indentation

shuang26 commented on src/main/java/rocky/command/Command.java:
```diff
@@ -0,0 +1,88 @@
+ * Class to capture fields and encapsulate behavior of a Command
+ * A Command is parsed from user input
+ */
+public class Command {
```
Great use of a command class to separate different user inputs.

shuang26 reviewed (commented):
Overall, I found your code easy to read and the documentation is done well.

flljy940 committed:
Configure gradle for checkstyle

flljy940 committed:
Add JavaFX GUI

flljy940 committed:
Fix minor issue

flljy940 committed:
Update ParserTest

flljy940 committed:
Add assertions for non-empty response string builders

flljy940 committed:
Refactor code to use guard statements and SLAP

flljy940 committed:
Extract constant and use guard statements

flljy940 committed:
Rename method

flljy940 committed:
Fix code quality and checkstyle issue

flljy940 committed:
Fix logging method and code quality

flljy940 committed:
test(parser): add test for parsing command of adding deadline

Added a new test case to ensure the parser for command of adding deadline task works.

flljy940 committed:
Merge pull request #1 from flljy940/branch-A-Assertions

Add assertions for non-empty response string builders

flljy940 committed:
Merge master into branch-A-CodeQuality to resolve conflicts

flljy940 committed:
Merge branch 'master' into branch-A-CodeQuality

flljy940 committed:
Merge pull request #2 from flljy940/branch-A-CodeQuality

Branch a code quality

flljy940 committed:
Add github actions workflow

flljy940 committed:
Delete I/O redirection tests

flljy940 committed:
Add streams

flljy940 committed:
Commit changes

flljy940 committed:
Streams feature

flljy940 committed:
Merge pull request #3 from flljy940/branch-A-Streams

Streams feature

flljy940 committed:
Merge branch 'master'
Merge branch 'master' of https://github.com/flljy940/ip

flljy940 committed:
Implement fuzzy search for Tasks

The command "find" used to search for task uses a fuzzy string matching
algorithm using Levenshtein distances between strings to calculate
similarities.

The chosen strategy of matching is a partial match, as to provide a
substring search as well. The output of the algorithm is between 0 and
100, and 75 is chosen to be the threshold as it was experimented to be
the most forgiving while at the same time requires a "close enough"
match.

This feature enables users to search tasks with approximate string matching, improving search flexibility.

flljy940 committed:
Merge branch 'master' into branch-C-BetterSearch

flljy940 committed:
Merge pull request #4 from flljy940/branch-C-BetterSearch

Implement fuzzy search for Tasks

flljy940 committed:
Fix merge conflicts

flljy940 committed:
Fix dependencies issues

flljy940 committed:
Fix issue of gui not closing after bye command

flljy940 committed:
Change to wider window

flljy940 committed:
Change dialog box layout and color, and recolor main window

flljy940 committed:
Set theme

flljy940 committed:
Add window title

flljy940 committed:
Set theme jekyll-theme-slate

flljy940 committed:
Fix javafx issue

flljy940 committed:
Update MainWindow

flljy940 committed:
Add UI screenshot for github pages

flljy940 committed:
Update user guide

flljy940 committed:
Add acknowledgement

flljy940 committed:
Update user guide

flljy940 committed:
Fix FAQ

flljy940 committed:
Delete expected output

flljy940 committed:
Fix iP code quality

flljy940 committed:
Refactor handleActionAndRespond

flljy940 committed:
Follow coding standard to header comments
